go/vt/vttablet/tabletmanager/vreplication/engine.go

```diff
@@ -266,6 +268,19 @@ func (vre *Engine) initControllers(rows []map[string]string) {
 	}
 }
 
+// Changes "MASTER" into "PRIMARY" and writes "PRIMARY" back into the database
+func (vre *Engine) repairRow(row map[string]string) {
+	oldVal := row["tablet_types"]
+	row["tablet_types"] = strings.ToUpper(row["tablet_types"])
+	row["tablet_types"] = strings.Replace(row["tablet_types"], "MASTER", "PRIMARY", -1)
+	if row["tablet_types"] != strings.ToUpper(oldVal) {
+		query := fmt.Sprintf("update _vt.vreplication set tablet_types=%v where id=%v", row["tablet_types"], row["id"])
+		if _, err := vre.dbClientFactoryFiltered().ExecuteFetch(query, 1); err != nil {
+			log.Errorf("could not repair row: %v: %v", row, err)
+		}
+	}
+}
+
```

Comment:
I changed a couple of test cases in `engine_test.go` to verify tablet_types returned from the select query, but whether I put `master,replica`, `MASTER,REPLICA`, or `PRIMARY,REPLICA`, they all pass. What is the right result to expect here?

Reply:
This is not clear to me: can you paste what you tried?

Reply:
I made your suggested change to `TestEngineOpen()`, and for this line:
```
dbClient.ExpectRequest("select * from _vt.vreplication where db_name='db'", sqltypes.MakeTestResult(
		sqltypes.MakeTestFields(
			"id|state|source|tablet_types",
			"int64|varchar|varchar|varbinary",
		),
		fmt.Sprintf(`1|Running|keyspace:"%s" shard:"0" key_range:{end:"\x80"}|master,replica`, env.KeyspaceName),
	), nil)
```
if the `master,replica` is `MASTER,REPLICA` or `PRIMARY,REPLICA`, all pass.

Reply:
That is fine. Here you are defining the result returned by the mock database layer for this query. The value you are putting  is what the `update` query will see in the `tablet_types` column. I had added this test because the last iteration was generating an invalid repair query. Since this query only ran when there was `master` in the tablet_type, without this test it was not getting called. With the current implementation this test is no longer as relevant since we always execute the query as part of WithDDL.